lib/jormungandr/src/Cardano/Wallet/Jormungandr/Binary.hs

```diff
@@ -258,9 +263,15 @@ getTransaction = label "getTransaction" $ do
 
 putSignedTx :: (Tx, [TxWitness]) -> Put
 putSignedTx (tx@(Tx inputs outputs), witnesses) = withSizeHeader16be $ do
+    unless (length inputs < 256) $
+        fail "number of inputs must be lower than 256"
+    unless (length outputs < 256) $
+        fail "number of outputs must be lower than 256"
+    unless (length inputs == length witnesses) $
+        fail "number of witnesses must equal number of inputs"
     putWord8 2
-    putWord8 $ fromIntegral $ length inputs
-    putWord8 $ fromIntegral $ length outputs
+    putWord8 $ overflowSafeCoerce $ length inputs
+    putWord8 $ overflowSafeCoerce $ length outputs
```

Comment:
We should either check the length with `unless` or with `overflowSafeCoerce`. Doing both is overkill.